### PR TITLE
Add Textfield This closes #239 and #252

### LINKED ISSUE #239: Add Textfield to Bit.Client.Web.BlazorUI
Add Textfield component base on https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield document

### PR DESCRIPTION
**Pull request by mirza-developer**
1- Implement BitTextField
2- Write and run test
3- Fix on focus move

### CODE REVIEW

**Review by hamedhajiloo — COMMENTED** [review]
.

**Comment by msynk** [comment]
based on our conversation:
- undo changes on the project file
- merged with the latest version source code
- format .razor code
- change style and CSS class assignments based on the latest version of source code

File: src/Client/Web/Bit.Client.Web.BlazorUI/Bit.Client.Web.BlazorUI.csproj [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="5.0.3" />
+    <PackageReference Include="Delegate.SassBuilder" Version="1.4.0" />
+    <PackageReference Include="Microsoft.TypeScript.MSBuild" Version="4.1.4">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
 
-</Project>
+</Project>
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Bit.Client.Web.BlazorUI.csproj:20]: please do not apply any changes to project files.
simply undo these changes.

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/_imports.razor [hunk 2]
```diff
@@ -1,2 +1,2 @@
 ﻿@using Bit.Client.Web.BlazorUI;
-@using Microsoft.AspNetCore.Components.Web;
+@using Microsoft.AspNetCore.Components.Web;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI.Tests/_imports.razor:2]: please undo this change.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor.cs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                    case nameof(OnClick):
+                        OnClick = (EventCallback<MouseEventArgs>)parameter.Value;
+                        break;
+                }
+            }
+            return base.SetParametersAsync(parameters);
+        }
+    }
+
+    public enum TextFieldType
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor.cs:166]: please move this enum into its own separate file.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor [hunk 4]
```diff
@@ -0,0 +1,40 @@
+﻿@namespace Bit.Client.Web.BlazorUI
+@inherits BitComponentBase
+<div @ref="Element"
+     style="@GetElementStyle()"
+     class="@GetElementClass() @FocusClass">
+    @if (IsMultiLine)
+    {
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor:7]: It's also required to check if (Type == TextFieldType.Text) in order to render a textarea.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        [Parameter] public string Placeholder { get; set; }
+        [Parameter] public bool IsReadonly { get; set; } = false;
+        [Parameter] public TextFieldType Type { get; set; } = TextFieldType.Text;
+        [Parameter]
+        public bool IsMultiLine
+        {
+            get => _isMultiLine;
+            set
+            {
+                if (Type == TextFieldType.Text)
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor.cs:22]: This check is not required, it's just a property setter and should not be conditional.

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Index.razor [hunk 6]
```diff
@@ -1,9 +1,16 @@
...
+    private string _showPasswordText = "show";
     private void MessageBarButton(MouseEventArgs args)
     {
         _isMessageBarHidden = true;
     }
+    private void ChangeInputType(MouseEventArgs args)
+    {
+        _inputType = _inputType == TextFieldType.Text ? TextFieldType.Password : TextFieldType.Text;
+        _showPasswordText = _inputType == TextFieldType.Text ? "hide" : "show";
+    }
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Index.razor:13]: move the @code content into the codebehind file (Index.razor.cs)

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/_Imports.razor [hunk 7]
```diff
@@ -8,4 +8,4 @@
 @using Bit.Client.Web.BlazorUI.Playground.Web;
 @using Bit.Client.Web.BlazorUI.Playground.Web.Shared;
 @using System.Net.Http.Json;
-@using Bit.Client.Web.BlazorUI;
+@using Bit.Client.Web.BlazorUI;
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/_Imports.razor:11]: undo this change.

**Review by msynk — CHANGES_REQUESTED** [review]
please resolve the comments

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor.cs [hunk 8]
```diff
@@ -0,0 +1,158 @@
+﻿using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+using System.Threading.Tasks;
+
+namespace Bit.Client.Web.BlazorUI
+{
+    public partial class BitTextField
+    {
+        private bool _isMultiLine = false;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitTextField.razor.cs:9]: This line should be removed

**Review by msynk — CHANGES_REQUESTED** [review]

**Pull request merged**